Library/Homebrew/formula_installer.rb

```diff
@@ -1001,14 +1001,48 @@ def link(keg)
     @show_summary_heading = true
   end
 
+  sig { params(service_hash: Hash).String }
+  def generate_plist(service_hash)
```

Comment:
Think having this live in a new file e.g. `service.rb` or something would be preferable to `formula_installer`